third_party/proton/proton/hooks/hook.py

```diff
@@ -103,10 +103,10 @@ def register(hook: Hook, session: int) -> None:
 
     @staticmethod
     def unregister(session: Optional[int] = None) -> None:
-        if session and session not in HookManager.session_hooks:
+        if session is not None and session not in HookManager.session_hooks:
             return
 
-        if not session:
+        if session is None:
```

Comment:
Previously we will mistakenly clean up all hooks when only `session_id = 0` is passed but there could exist two sessions